src/stan-users-guide/regression.Rmd

```diff
@@ -25,24 +27,15 @@ $$
 y_n \sim \operatorname{normal}(\alpha + \beta X_n, \, \sigma).
 $$
 
-This latter form of the model is coded in Stan as follows.
+This latter form of the model is coded in Stan as follows [linear_regression_1](https://github.com/stan-dev/docs/tree/footnote/src/stan-users-guide/linear_regression_1)
 
-```
-data {
-  int<lower=0> N;
-  vector[N] x;
-  vector[N] y;
-}
-parameters {
-  real alpha;
-  real beta;
-  real<lower=0> sigma;
-}
-model {
-  y ~ normal(alpha + beta * x, sigma);
-}
-```
 
+```{r echo=FALSE}
+library(knitr)
+library(readr)
+stan_file = read_file('linear_regression_1/regression_1.stan')
+writeLines(stan_file)
+```
```

Comment:
Changed to print_file() from before-chapter.R--pushed

Directory name 'linear_regression_1' reflects heading name for sub-section, '## Linear Regression' . This seems the easiest way for readers to track what dir they want. The Regression.Rmd covers a range of topics, e.g., QR Decomposition which would get a directory 'qr_decomposition_n' where n increments based on order in the chapter. 

If there is need for multiple implementations in the directory I think they can whatever file name they want. Looking ahead in the comments I think I'd like to to have something like
'linear_regression_naive_coding.stan' and 'linear_regression_efficient_coding.stan' for this directory. The interface programs that call the stan code may or may not be 1-1 so 'linear_regression.R' may have calls to both .stan models. 

Not sure I like the numeric suffix but they function like figure numbers which I think is helpful.

Reply:
I think Andrew makes a good point that the footnotes can stay in place while chapters, sections, etc. get reorganized.  

> My inclination would be to have each "footnote" in a mini-directory that has a descriptive name (e.g., SImple_Linear_Regression), rather than being tied to chapters of the manual.  My concern about being tied to chapters of the manual is that I could imagine the manual evolving with chapters split in two, etc.

Reply:
`Simple_Linear_Regression` uses both camel case and underscore separators.  We don't need both.  So choose either `SimpleLinearRegression` or `simple_linear_regression`.  I strongly prefer the latter.  I like `simple-linear-regression` with hyphens even more.

Reply:
Ok, I'll drop the suffix numbering scheme and keep directory names descriptive. 

I'd like one naming scheme that works across .stan, .R and .py programs based on legal variable names. Stan/R doesn't allow for '-' and camel case does not seem adopted in R so that leaves us with '_'.  Everyone ok with that?

Reply:
Yes, I'm OK with that.

P.S.  `-` works in file names in all of those languages.  As do file names starting with integers.  Where we've run into problems is when interfaces automatically promote file names to identifiers.